File: VB/15_CommentAndNote/GetSlideComments.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation

Namespace GetSlideComments
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PPT document
			Dim presentation As New Presentation()

			'Load document from disk
			presentation.LoadFromFile("..\..\..\..\..\..\Data\Comments.pptx")

			'Loop through comments
			For Each commentAuthor As ICommentAuthor In presentation.CommentAuthors
				For Each comment As Comment In commentAuthor.CommentsList
					'Get comment information
					Dim commentText As String = comment.Text
					Dim authorName As String = comment.AuthorName
					Dim time As Date = comment.DateTime
					MessageBox.Show("Comment text : " & comment.Text & vbLf & "Comment author : " & comment.AuthorName & vbLf & "Posted on time : " & comment.DateTime)
				Next comment
			Next commentAuthor
		End Sub
	End Class
End Namespace

File: CS/15_CommentAndNote/GetSlideComments.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;

namespace GetSlideComments
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PPT document
            Presentation presentation = new Presentation();

            //Load document from disk
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\Comments.pptx");

            //Loop through comments
            foreach (ICommentAuthor commentAuthor in presentation.CommentAuthors)
            {
                foreach (Comment comment in commentAuthor.CommentsList)
                {
                    //Get comment information
                    string commentText = comment.Text;
                    string authorName = comment.AuthorName;
                    DateTime time = comment.DateTime;
                    MessageBox.Show("Comment text : "+ comment.Text +"\n"+"Comment author : " + comment.AuthorName + "\n" + "Posted on time : " + comment.DateTime);
                }
            }
        }
    }
}